x/accounts/internal/implementation/api_builder.go

```diff
@@ -0,0 +1,106 @@
+package implementation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var (
+	errNoInitHandler    = errors.New("no init handler")
+	errNoExecuteHandler = errors.New("account does not accept messages")
+	ErrInvalidMessage   = errors.New("invalid message")
+)
+
+// NewInitBuilder creates a new InitBuilder instance.
+func NewInitBuilder() *InitBuilder {
+	return &InitBuilder{}
+}
+
+// InitBuilder defines a smart account's initialisation handler builder.
+type InitBuilder struct {
+	// handler is the handler function that will be called when the smart account is initialized.
+	// Although the function here is defined to take an interface{}, the smart account will work
+	// with a typed version of it.
+	handler func(ctx context.Context, initRequest interface{}) (initResponse interface{}, err error)
```

Comment:
nit: if you use `any` over `interface{}` it shortens the code and makes it easier to read

Reply:
good point :)